### DISCUSSION TITLE
How to skip `__typename` in GraphQL operations?

### DISCUSSION THREAD

**Discussion by todorone** [urql-graphql/urql#2789] [Q&A] [answered]
I use `@urql/core` as a GraphQL operations engine w/o other urql components like caching.

I want to skip automatic adding of `__typename` property to queries(for my case it's critical to have as lean request and response payloads as possible) but didn't find any solution how to achieve it with `@urql/core`. Any help is appreciated. 🙏

**Comment by JoviDeCroock** [comment, marked as answer]
We are only adding `__typename` automatically when you are using the `cacheExchange` so if you want to remove that you will have to create your client like this:

```js
import { createClient, dedupExchange, fetchExchange } from "@urql/core";

const client = createClient({ url: '', exchanges: [dedupExchange, fetchExchange] })
```

The method that adds these is [`formatDocument`](https://github.com/FormidableLabs/urql/blob/main/packages/core/src/utils/typenames.ts#L66) which is only used in the [`cacheExchange`](https://github.com/FormidableLabs/urql/blob/main/packages/core/src/exchanges/cache.ts#L11)

**Reply by kitten** [reply]
On a side note, adding introspected fields is extremely important for caching and it really shouldn't be a concern for making requests lean. What I mean is that caching over time as part of state management and requests in general, be it via caching on the server-side or clientside, should become more important than saving a couple of bytes in the response 😅

So, especially since it's often in large responses repetitive which compression takes care of, I really wouldn't recommend defeating it, since it prevents any automatic caching from working properly

**Reply by todorone** [reply]
@kitten i totally agree whatever You said on every common case, i have pretty edge case though.

It's React Native app targeting Android TV devices which are really low spec - like 5yo Android low-end phone with 1-2Gb of memory, so network payload matters not just because of traffic but because of slow JSON parsing.

To improve performance on programs schedule screen(a lot of dynamic fetching when user moves around) we use manual case-tailored caching and removing `__typename` from payloads may improve performance by something like 1-5% which is big for this case.

**Reply by todorone** [reply]
@JoviDeCroock Thanks a lot for prompt reply.

~~I've tried the way of client creation You provided, but it didn't help me.~~

~~I still see `__typename` tags present in `query` property of POST request that `urql` performs. Just for clarity, i want that `urql`(if it's possible) omits `__typename` in query, rather than removing them on response processing step.~~

UPDATE: Sorry, it was caching issue. It works, thanks again! 💙
